crates/abq_cli/tests/cli.rs

```diff
@@ -5171,3 +5574,446 @@ fn retry_continued_manifest_read_on_worker_death() {
+                OpaqueWrite(pack(RawTestResultMessage::fake(proto))),
+            ],
+        },
+        Exit(0),
+    ];
+    let packed = pack_msgs_to_disk(simulation);
+
+    let test_args = {
+        let simulator = native_runner_simulation_bin();
+        let simfile_path = packed.path.display().to_string();
+        let args = vec![format!("test"), format!("--worker=1"), format!("-n=1")];
+        let mut args = conf.extend_args_for_client(args);
+        args.extend([s!("--"), simulator, simfile_path]);
+        args
+    };
+
+    let CmdOutput {
+        stdout,
+        stderr,
+        exit_status,
+    } = Abq::new(format!("{name}_test"))
+        .args(test_args)
+        .env([
+            ("CI", "true"),
+            ("GITHUB_RUN_ID", "test-run-id"),
+            ("ABQ_API", &server.url()),
+            ("RWX_ACCESS_TOKEN", &access_token.to_string()),
+        ])
+        .run();
+    assert!(
+        exit_status.success(),
+        "STDOUT:\n{stdout}\nSTDERR:\n{stderr}"
+    );
+    assert!(
+        stdout.contains("1 tests, 0 failures"),
+        "STDOUT:\n{stdout}\nSTDERR:\n{stderr}"
+    );
+
+    let report_args = {
+        let args = vec![
+            format!("report"),
+            format!("--reporter=dot"),
+            format!("--color=never"),
+        ];
+        conf.extend_args_for_client(args)
+    };
+
+    let CmdOutput {
+        stdout,
+        stderr,
+        exit_status,
+    } = Abq::new(name.to_string() + "_report")
+        .args(report_args)
+        .env([
+            ("CI", "true"),
+            ("GITHUB_RUN_ID", "test-run-id"),
+            ("ABQ_API", &server.url()),
+            ("RWX_ACCESS_TOKEN", &access_token.to_string()),
+        ])
+        .run();
+
+    assert!(
+        exit_status.success(),
+        "STDOUT:\n{stdout}\nSTDERR:\n{stderr}"
+    );
+    assert!(
+        stdout.contains("1 tests, 0 failures"),
+        "STDOUT:\n{stdout}\nSTDERR:\n{stderr}"
+    );
+
+    // Check that the run ID also exists on the queue explicitly.
+    {
+        let report_args = {
+            let args = vec![
+                format!("report"),
+                format!("--reporter=dot"),
+                format!("--color=never"),
+                format!("--queue-addr={queue_addr}"),
+                format!("--run-id={in_run_id}"),
+            ];
+            conf.extend_args_for_client(args)
+        };
+
+        let CmdOutput {
+            stdout,
+            stderr,
+            exit_status,
+        } = Abq::new(name.to_string() + "_report")
+            .args(report_args)
+            .run();
+
+        assert!(
+            exit_status.success(),
+            "STDOUT:\n{stdout}\nSTDERR:\n{stderr}"
+        );
+        assert!(
+            stdout.contains("1 tests, 0 failures"),
+            "STDOUT:\n{stdout}\nSTDERR:\n{stderr}"
+        );
+    }
+}
+
+#[test]
+#[with_protocol_version]
+#[serial]
+fn test_with_org_access_token_and_explicit_cli_run_id() {
```

Comment:
This exercises ORG + run ID from CLI arg => always use result of remote configuration